### PR TITLE
[1.2.x] Throw IllegalStateException when undeploying nonexistent app

### LINKED ISSUE #152: Throw IllegalStateException when undeploying the app that has not been deployed
This is based on https://github.com/spring-cloud/spring-cloud-deployer/issues/185

### PR DESCRIPTION
**Pull request by ilayaperumalg**
Resolves #152

### CODE REVIEW

**Comment by ilayaperumalg** [comment]
This PR is based on https://github.com/spring-cloud/spring-cloud-deployer/pull/188

File: src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java [hunk 1]
```diff
@@ -161,6 +165,13 @@ private void assertApplicationDoesNotExist(String deploymentId, AppStatus status
 		}
 	}
 
+	private void assertApplicationExists(String deploymentId, AppStatus status) {
+		DeploymentState state = status.getState();
+		if (state == DeploymentState.unknown) {
+			throw new IllegalStateException(String.format("App %s is not deployed", deploymentId));
```
Comment (trisberg) [inline src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java]: When merging update message to say ""App %s is not in a deployed state.""

**Comment by trisberg** [comment]
Don't merge until master is on 1.2.0.BUILD-SNAPSHOT

**Comment by trisberg** [comment]
Fixed deployer tests and deployer-spi version; merged as commit 9edb8484df6da55b27288c913a1c324460347b04

**Pull request closed without merging**